src/shared_modules/utils/rsaHelper.hpp

```diff
@@ -93,7 +92,7 @@ class RSAHelper final
         DEFER([&]() { T::RSA_free(rsa); });
 
         // Decrypt the ciphertext using RSA private key
-        const auto decryptedLen = T::RSA_private_decrypt(RSA_SIZE,
+        const auto decryptedLen = T::RSA_private_decrypt(input.size(),
```

Comment:
From the [openssl documnetation](https://www.openssl.org/docs/man1.1.1/man3/RSA_private_decrypt.html):

> flen must not be more than RSA_size(rsa) - 11 for the PKCS #1 v1.5 based padding modes, not more than RSA_size(rsa) - 42 for RSA_PKCS1_OAEP_PADDING and exactly RSA_size(rsa) for RSA_NO_PADDING. When a padding mode other than RSA_NO_PADDING is in use, then RSA_public_encrypt() will include some random bytes into the ciphertext and therefore the ciphertext will be different each time, even if the plaintext and the public key are exactly identical. The returned ciphertext in to will always be zero padded to exactly RSA_size(rsa) bytes. to and from may overlap.

We have to be careful with the size we provide to this method based on the padding

Reply:
I'd add tests for this change.